doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,247 @@
+  - This is essential so that host/PMon container can determine the vendor name from eeprom and import the module of the transceiver appropriately
+
+
+### Port to module mapping
+- Another thing that is required is once we do have a vendor to module mapping, we need to map appropriate port to a module as well
+
+#### Background
+
+  - Basically, the key idea is once we have a port identified as being of a certain vendor and it has been identified to load a certain module which we can gather from the mapping described above we still need to call the correct module on each port each time we call the API on the port
+  - want to maintain this mapping in memory since xcvrd does not want to read/parse this y_cable_vendor_mapping.csv file again and again each time we call the Y-cable API
+  - Also note that the module loaded might change during xcvrd running lifetime since cable can be changed from one vendor to another. So we need to take this into consideration as well
+
+#### Proposed Solution
+
+  - Each module of the Y cable vendor can be a class (of each transceiver type) and all we need to do is instantiate the objects of these classes as class instances and these objects will provide the interface of calling the API's for the appropriate vendor Y cable.
+  - This instantiation will be done inside xcvrd, when xcvrd starts
+  - These objects basically emulate Y cable instances and whatever action/interaction needs to be done with the Ycable the methods of these objects would provide that
+  - each vendor in their implementation can inherit from a base class where there will be definitions for all the supported capabilities of the Y-cable.
+  - If the vendor does not support or implement the utility, it will just raise a not implemented error from the base class itself
+
+For example a typical module of the vendor can be like this
+```
+
+	class Ycable(Ycablebase):
+            #all vendor modules inherit from Ycablebase
+	    def __init__(self):
+
+	    def toggle_mux_to_torA(self, port):
+		#implement or raise exception if not implemented
+		raise NotImplementedError
+	    def toggle_mux_to_torB(self, port):
+		#implement or raise exception if not implemented
+		raise NotImplementedError
+
+	    def check_prbs(self, port):
+		#implement or raise exception if not implemented
+		raise NotImplementedError
+
+	    def only_credo(self, port):
+		raise NotImplementedError
+
+	    def only_amphenol(self, port):
+		raise NotImplementedError
+
+```
+the base class would be like this
+
+```
+	class Ycablebase(object):
+	    def __init__(self, port):
+		self.port = port
+
+	    def toggle_mux_to_torA(self, port):
+		raise NotImplementedError
+	    def toggle_mux_to_torB(self, port):
+		raise NotImplementedError
+
+	    def check_prbs(self, port):
+		raise NotImplementedError
+
+	    def only_credo(self, port):
+		raise NotImplementedError
+
+	    def only_amphenol(self, port):
+		raise NotImplementedError
+
+```
```

Comment:
Suggest moving the base class definition example above the vendor subclass one.

Reply:
fixed